extensions/wikia/MultiTasks/MultiDeleteController.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare( strict_types=1 );
+
+use Wikia\Logger\Loggable;
+use User;
+
+final class MultiDeleteController extends WikiaController {
+	use Loggable;
+
+	public function allowsExternalRequests() {
+		return false;
+	}
+
+	public function multiDeletePages() {
+		$this->response->setFormat( WikiaResponse::FORMAT_JSON );
+		if ( !$this->request->wasPosted() ) {
+			$this->response->setCode( 405 );
+			return;
+		}
+		$pagesToDelete = $this->getVal( 'pages' );
+		$userId = $this->getVal( 'user' );
+		$reason = $this->getVal( 'reason' );
+		if ( empty( $pagesToDelete ) || empty( $userId ) ||  empty( $reason ) ) {
+			$this->response->setCode( WikiaResponse::RESPONSE_CODE_BAD_REQUEST );
+			return;
+		}
+		$user = User::newFromId( $userId );
+
+		foreach ( $pagesToDelete as $pageName ) {
+			$page = WikiPage::factory( \Title::newFromText( $pageName ) );
+
+			if ( $page !== null ) {
+				$e = '';
+				$page->doDeleteArticle( $reason, false, null, null, $e, $user );
```

Comment:
Two issues here:
* `null` is passed as article id, mixing up types; this should be 0
* also the boolean `$commit` value is set to null which is wrong for two reasons; first it is also mixing up the types; second, passing an empty value will probably start a new transaction but will not commit it, so the article won't be deleted